src/main/java/com/ibm/cics/bundle/deploy/BundleDeployHelper.java

```diff
@@ -81,7 +87,22 @@ public static void deployBundle(URI endpointURL, File bundle, String bunddef, St
 		HttpEntity httpEntity = mpeb.build();
 		httpPost.setEntity(httpEntity);
 		
-		HttpClient httpClient = HttpClientBuilder.create().build();
+		HttpClient httpClient;
+		if (!allowSelfSignedCertificate) {
+			httpClient = HttpClientBuilder.create().build();
+		} else {
+			try {
+				httpClient = HttpClients.custom()
+						.setSslcontext(new SSLContextBuilder().loadTrustMaterial(null, (chain, type) -> true).build())
+						.setHostnameVerifier(new AllowAllHostnameVerifier())
+						.build();
+			} catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e) {
+				throw new BundleDeployException("Error instantiating secure connection", e);
```

Comment:
Is this message inverted? Doesn't seem like a very secure connection 😆

Reply:
Hmm. No it was intentional. It's a secure connection because it's using SSL and we had an error instantiating the connection.

The fact that it's less secure than it could have been....?

We could just say error instantiating SSL connection.

Reply:
Yeah, if that doesn't add more confusion because it's almost certainly TLS... I'm happy with it as it stands.

Reply:
We could say "error instantiating encrypted connection". But I think I prefer secure! :)